lib/three_scale/tenant_id_integrity_checker.rb

```diff
@@ -92,7 +92,7 @@ def last_table_alias_from_sql(sql)
     def models_with_tenant_id
       Rails.autoloaders.main.eager_load_dir("#{Rails.root}/app/models")
       all_models = ApplicationRecord.descendants.select(&:arel_table).reject(&:abstract_class?)
-      all_models.select! { _1.attribute_names.include? "tenant_id" }
+      all_models.select! { _1.attribute_names.include? "tenant_id" }.sort_by(&:name) || []
```

Comment:
Depending on the order of the models, the checker gives different results (because it skips associations, for which the inverse has already been checked).

Reply:
I added a commit with a better fix for it. The sorting btw introduced another issue - that child STI models could be processed before the base STI model, leading to missed inconsistencies. I fixed it anyway but also changed the tests to not depend on order of classes being loaded.